fabric-biome-api-v1/src/main/java/net/fabricmc/fabric/mixin/biome/BiomeSourceMixin.java

```diff
@@ -33,7 +33,7 @@
 public class BiomeSourceMixin {
 	@Redirect(method = "getBiomes", at = @At(value = "INVOKE", target = "Ljava/util/function/Supplier;get()Ljava/lang/Object;"))
 	private Object getBiomes(Supplier<Set<RegistryEntry<Biome>>> instance) {
-		var biomes = new HashSet<>(instance.get());
+		var biomes = new LinkedHashSet<>(instance.get());
 		fabric_modifyBiomeSet(biomes);
 		return Collections.unmodifiableSet(biomes);
```

Comment:
As I said on discord, I think we should change this to just retrun the original set when no modifications have taken place. In TheEndBiomeSourceMixin you would end up with something along the lines of:

![Screenshot 2023-10-07 at 14 11 05](https://github.com/FabricMC/fabric/assets/4324090/6205a780-df6c-4aa5-9b37-8b88b5218c2b)